arangod/Pregel/Conductor.cpp

```diff
@@ -730,7 +687,6 @@ template<typename OutType, typename InType>
 auto Conductor::_sendToAllDBServers(std::string const& path,
                                     InType const& message)
     -> ResultT<std::vector<OutType>> {
-  _callbackMutex.assertLockedByCurrentThread();
```

Comment:
Is this ok? Or will it be ok once we have all the future-types sorted out?

Reply:
It should be ok when the future-types are all in place. I had problems running pregel when this was not deleted and had problems acquiring the lock and keep it for that long with the new future.

Reply:
I think this particular problem will be solved by us using `Guarded` wrappers around the datastructures we affect later; keep in mind how this callbackMutex seems to have come to be: only one thread is supposed to be in the "critical Section" of the conductor...